vendor/assets/javascripts/react-chosen.js

```diff
@@ -0,0 +1,69 @@
+(function(window, React, $) {
```

Comment:
It's not about the versions, it's about the fact that the test helpers are in React.addons, which gets pulled in by react_rails if you say so in the Rails config. With the rails-assets version it gets pulled in if you `//= require react_with_addons`. We want this only for the test environment (so far). The react_rails variant is obviously more convenient, since it is easier to deal with the environments from the Ruby side of things. If we'd have a way to ignore the rails-assets-react dependency, that would be fine. But, as I said, it's not even appearing in the assets load paths. The actual problem is having two different sources with separate dependency trees. This is not limited to react and may or may not fall onto our feet without warning, again and again.

I'll do a little more digging, though. Maybe I can at least get rid of the vendor file...